### PR TITLE
Update risk.py

### PR DESCRIPTION
**Pull request by drozzy**
Fixing #63

### CODE REVIEW

File: simglucose/analysis/risk.py [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+    HBGI = np.mean([r[1] for r in risks])
+    RI = np.mean([r[2] for r in risks])
+
     return (LBGI, HBGI, RI)
+
+def risk(BG):
+    MIN_BG = 20.0
+    MAX_BG = 600.0
+    if BG <= MIN_BG: 
+        return (100.0, 0.0, 100.0)
```
Comment (jinyu-loopmind) [inline simglucose/analysis/risk.py:17]: It is not intuitive based on the formula below that why the risk index is saturated at 100 at MIN_BG and MAX_BG. Can you add some comments explaining this?

Comment (jinyu-loopmind) [reply simglucose/analysis/risk.py:17]: (Sorry, I was using my other github account instead of the jxx123 account.)

**Review by jinyu-loopmind — COMMENTED** [review]
In general, this is a big change on how to compute risk index. It might have big impact on people who are already used to the old scales, especially the risk index is no longer the sum of LBGI and HBGI. Risk index was computed as the sum of LBGI and HBGI for a long time in the original UVa/Padova Simulator. 

To minimize the impact, I am thinking about introducing your new risk index computation as an alternative instead of completely replacing them. Maybe a more gentle way to do this is to issue deprecation warning, and advice people to use the new risk index instead. Let me know what you think on this. Thanks!

**Comment by drozzy** [comment]
Do you know where in UVa/Padova Simulator they define it? I would be curios to see their implementation.

Regarding current use, I  can't imagine people have used it because it produces wrong output at large values.

For example, here is the current risk if I let the glucose rise indefinintely:
original_extreme_values

As you can see the function just becomes undefined at some point and gets set to zero.

Here is the proposed risk function for the same trajectory:
proposed_extreme_values


Regarding compatibility, the new risk behaves the same for values in the acceptable range.
Here is the current risk:
original_risk

and here is proposed risk:
proposed_risk

I welcome more testing and feedback of my proposal, but believe it is especially important for automated optimizers, where such a drop may lead to incorrect learned behavior.

File: simglucose/analysis/risk.py (hunk 1 above)
Comment (drozzy) [reply simglucose/analysis/risk.py:17]: Those are just the values to which the risk formula was fit. In other words they aimed to make the risk maximum when it was either 20 or 600.
See this paper: https://diabetesjournals.org/care/article/20/11/1655/21162/Symmetrization-of-the-Blood-Glucose-Measurement

In the appendix they derive risk as follows. Note that the units are different mmol/l - but in mg/dl these limits are 20 and 600:
Screenshot_20231003_110426

Comment (jxx123) [reply simglucose/analysis/risk.py:17]: Thanks for the explanation. Can you add a brief comment in the code about this saturation just for code readability? Thanks!

**Comment by jxx123** [comment]
> Do you know where in UVa/Padova Simulator they define it? I would be curios to see their implementation.
> 
> Regarding current use, I can't imagine people have used it because it produces wrong output at extreme values.
> 
> For example, here is the current risk if I let the glucose fall indefinitely: original_extreme_values
> 
> As you can see the function just becomes undefined at some point and gets set to zero.
> 
> Here is the proposed risk function for the same trajectory: proposed_extreme_values
> 
> Regarding compatibility, the new risk behaves the same for values in the acceptable range. Here is the current risk: original_risk
> 
> and here is proposed risk: proposed_risk
> 
> For large glucose values, the current risk is not a big problem as it does increase in the desired direction, however it is not bounded and goes to infinity: large_gluc_original_risk
> 
> the proposed risk caps this at 100: large_gluc_proposed_risk
> 
> I welcome more testing and feedback of my proposal, but believe it is especially important for automated optimizers, where such a drop may lead to incorrect learned behavior.

Wow, thank you so much for the detailed and compelling analysis. Yes, the current risk computation is definitely problematic. 

FYI, here is how the original UVa/Padova Simulator implements the risk index (they sum up LBGI and HBGI), where `i` means i-th patient. 
Screenshot by Dropbox Capture

I think I will approve your change anyway since the existing one is definitely wrong. Thank you so much!

File: simglucose/analysis/risk.py (hunk 1 above)
Comment (jxx123) [reply simglucose/analysis/risk.py:17]: Echo the latter comment, I will approve your change once this point is addressed.

**Comment by jxx123** [comment]
> Out of curiosity, where did you find this code? Is it accessible somewhere?

I had it as in the 2008 UVa/Padova Simulator Academic version. You might be able to request a version from epsilon group https://tegvirginia.com/software/t1dms-2014/, but I heard stories that there was no reply from them.

File: simglucose/analysis/risk.py (hunk 1 above)
Comment (drozzy) [reply simglucose/analysis/risk.py:27]: I've added that to the comment, not sure if you saw.

**Pull request merged**